kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesCrudDispatcher.java

```diff
@@ -316,4 +327,63 @@ private int doCreate(String path, String s, String event) {
     }
     return HttpURLConnection.HTTP_OK;
   }
+
+  private MockResponse validateRequestBodyAndHandleRequest(String s, Supplier<MockResponse> mockResponseSupplier) {
+    HasMetadata h = toKubernetesResource(s);
+    if (h != null && !isResourceValid(h)) {
+      return getUnprocessableEntityMockResponse(s, h);
+    }
+    return mockResponseSupplier.get();
+  }
+
+  private MockResponse getUnprocessableEntityMockResponse(String s, HasMetadata h) {
+    String statusBody = getStatusBody(h, HTTP_UNPROCESSABLE_ENTITY);
+    if (statusBody == null) {
+      statusBody = s;
+    }
+    return new MockResponse().setResponseCode(HTTP_UNPROCESSABLE_ENTITY).setBody(statusBody);
+  }
+
+  private String getStatusBody(HasMetadata h, int code) {
+    String kind = Utils.getNonNullOrElse(h.getKind(), "Unknown");
+    Status status = new StatusBuilder().withStatus("Failure")
+      .withReason("Invalid")
+      .withMessage(kind + " is invalid")
+      .withNewDetails()
+      .withKind(h.getKind())
+      .withCauses(getFailureStatusCause(h))
+      .endDetails()
+      .withCode(code)
+      .build();
+    try {
+      return Serialization.jsonMapper().writeValueAsString(status);
+    } catch (IOException ioException) {
+      return null;
+    }
+  }
+
+  private StatusCause getFailureStatusCause(HasMetadata h) {
+    StatusCauseBuilder statusCauseBuilder = new StatusCauseBuilder();
+    if (h == null) {
+      statusCauseBuilder.withField(".")
+        .withMessage("Nothing provided")
+        .withReason("ValueRequired");
+    } else if (h.getMetadata() == null ||
+      (h.getMetadata().getName() == null && h.getMetadata().getGenerateName() == null)) {
+      statusCauseBuilder.withField(".metadata.name")
+        .withMessage("Required value: name or generateName is required")
```

Comment:
This is not OK.

isResourceValid should throw a checked Exceptionwith the given message (what happens if we add more validations in the future?).

The method `validateRequestBodyAndHandleRequest` should call `isResourceValid` or better renamed to `validateResource`. Whenever that method throws an exception, the failed body is returned with the message provided by that Exception.

Not sure if my description is clear enough. If not, we can discuss in a call.